static/js/hashchange.js

```diff
@@ -180,7 +180,15 @@ function do_hashchange_normal(from_reload) {
         case "#":
             show_default_view();
             break;
+        // TODO/compatibility: In order to make the reload.js logic work
+        // after upgrades for self-hosted systems, treat "#recent_topics"
+        // as "#recent" until one can no longer directly upgrade from
+        // Zulip 5.x to the current version.
```

Comment:
I'd move this comment inside the case block.

Reply:
Moved and updated to be a permanent comment about why we're keeping the logic for this old hash.